imagetagger/imagetagger/images/views.py

```diff
@@ -92,6 +92,16 @@ def index(request):
         image_count_agg=Count('images')
     ).select_related('team').prefetch_related('set_tags') \
         .order_by('-priority', '-time')
+    
+    # add annotated image count to each imageset
+    for imageset in imagesets:
+        images = Image.objects.filter(image_set=imageset)
+        annotations = Annotation.objects.filter(image__in=images,
+                                            annotation_type__active=True).select_related()
+        annotatedimages = set()
+        for annotation in annotations:
+            annotatedimages.add(annotation.image)
+        imageset.annotated_image_count = len(annotatedimages)
```

Comment:
Nice! FWIW I think you can somewhat shorten this with a Python generator statement:

```python
imageset.annotated_image_count = len(set((annotation.image for annotation in annotations)))
```

It's the same thing just written on one line using a generator statement thing.

Reply:
fixed!